listenbrainz/webserver/static/js/src/PinRecordingModal.tsx

```diff
@@ -0,0 +1,174 @@
+        this.handleError(error, "Error while pinning recording");
+      }
+    }
+  };
+
+  handleBlurbInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    event.preventDefault();
+    const input = event.target.value.replace(/\s\s+/g, " "); // remove line breaks and excessive spaces
+    if (input.length <= this.maxBlurbContentLength) {
+      this.setState({ blurbContent: input });
+    }
+  };
+
+  handleError = (error: string | Error, title?: string): void => {
+    const { newAlert } = this.props;
+    if (!error) {
+      return;
+    }
+    newAlert(
+      "danger",
+      title || "Error",
+      typeof error === "object" ? error.message : error
+    );
+  };
+
+  render() {
+    const { recordingToPin } = this.props;
+    const { blurbContent } = this.state;
+    const track_name = recordingToPin?.track_metadata?.track_name;
+    const artist_name = recordingToPin?.track_metadata?.artist_name;
+
+    return (
+      <div
+        className="modal fade"
+        id="PinRecordingModal"
+        tabIndex={-1}
+        role="dialog"
+        aria-labelledby="PinRecordingModalLabel"
+        data-backdrop="static"
+      >
+        <div className="modal-dialog" role="document">
+          <form className="modal-content">
+            <div className="modal-header">
+              <button
+                type="button"
+                className="close"
+                data-dismiss="modal"
+                aria-label="Close"
+              >
+                <span aria-hidden="true">&times;</span>
+              </button>
+              <h4 className="modal-title" id="PinRecordingModalLabel">
+                Pin This Recording to Your Profile
+              </h4>
+            </div>
+            <div className="modal-body">
+              <p>
+                Why are you pinning{" "}
+                <b>
+                  {" "}
+                  {track_name} - {artist_name}
+                </b>
+                ? (Optional)
+              </p>
+              <div className="form-group">
+                <textarea
+                  className="form-control"
+                  id="blurb-content"
+                  placeholder="Tell us why you love this recording!"
+                  value={blurbContent}
+                  name="blurb-content"
+                  onChange={this.handleBlurbInputChange}
+                  rows={4}
+                  style={{ resize: "vertical" }}
+                  spellCheck="false"
+                />
+              </div>
+              <small style={{ display: "block", textAlign: "right" }}>
+                {blurbContent?.length} / {this.maxBlurbContentLength}
+              </small>
+              <small>
+                Pinning this recording will replace any recording currently
+                pinned. <br />
+                <b>
+                  {track_name} by {artist_name}
+                </b>{" "}
+                will be unpinned from your profile in one week.
```

Comment:
(optional, not blocking the PR)

Do we know the actual time for this? Maybe we could show it on hover or something, it might be a good user experience.

Reply:
Sure, I appended the timestamp to the text, that way it's viewed more easily as opposed to having to hover.